lib/rubocop/cop/style/string_concatenation.rb

```diff
@@ -141,15 +141,18 @@ def corrected_ancestor?(node)
           node.each_ancestor(:send).any? { |ancestor| @corrected_nodes&.include?(ancestor) }
         end
 
-        def replacement(parts)
+        def replacement(parts) # rubocop:disable Metrics/MethodLength
```

Comment:
Is it possible to eliminate this disable comment with a change like the following?

```diff
diff --git a/lib/rubocop/cop/style/string_concatenation.rb b/lib/rubocop/cop/style/string_concatenation.rb
index bd21afd38..840e9bce7 100644
--- a/lib/rubocop/cop/style/string_concatenation.rb
+++ b/lib/rubocop/cop/style/string_concatenation.rb
@@ -141,27 +141,22 @@ module RuboCop
           node.each_ancestor(:send).any? { |ancestor| @corrected_nodes&.include?(ancestor) }
         end

-        def replacement(parts) # rubocop:disable Metrics/MethodLength
-          interpolated_parts =
-            parts.map do |part|
-              case part.type
-              when :str
-                str_node_replacement(part)
-              when :dstr
-                if part.children.all?(&:str_type?)
-                  str_node_replacement(part)
-                else
-                  contents_range(part).source
-                end
-              else
-                "\#{#{part.source}}"
-              end
+        def replacement(parts)
+          interpolated_parts = parts.map do |part|
+            case part.type
+            when :str
+              adjust_str(part)
+            when :dstr
+              part.children.all?(&:str_type?) ? adjust_str(part) : contents_range(part).source
+            else
+              "\#{#{part.source}}"
             end
+          end

           "\"#{handle_quotes(interpolated_parts).join}\""
         end

-        def str_node_replacement(node)
+        def adjust_str(node)
           single_quoted?(node) ? node.value.gsub(/(\\|")/, '\\\\\&') : node.value.inspect[1..-2]
         end
```

Reply:
Thanks, it works. Applied.